File: android/src/main/java/io/glassfy/capacitor/plugin/GlassfyPlugin.java
package io.glassfy.capacitor.plugin;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Glassfy")
public class GlassfyPlugin extends Plugin {

    @PluginMethod
    public void sdkVersion(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void initialize(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void offerings(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void skuWithIdentifier(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void login(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void logout(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void permissions(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void purchaseSku(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void restorePurchases(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void setLogLevel(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void setDeviceToken(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void setExtraUserProperty(PluginCall call) {
        call.unimplemented();
    }

    @PluginMethod
    public void getUserProperty(PluginCall call) {
        call.unimplemented();
    }
}

File: ios/Plugin/GlassfyPlugin.m
#import <Capacitor/Capacitor.h>
#import <Foundation/Foundation.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(GlassfyPlugin, "Glassfy",
           CAP_PLUGIN_METHOD(sdkVersion, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(initialize, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setLogLevel, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(offerings, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(permissions, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(skuWithId, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(skuWithIdAndStore, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(connectCustomSubscriber, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(connectPaddleLicenseKey, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setEmailUserProperty, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setDeviceToken, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setExtraUserProperty, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(getUserProperty, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(purchaseSku, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(restorePurchases, CAPPluginReturnPromise);)